### PR TITLE
Add and Use Duplicate Detection Feature Flag

### PR DESCRIPTION
**Pull request by NicholasFlamy**
So I created the `DUPLICATE_DETECTION` feature flag and applied it to the places that it could be used based on the discussion is #8228.

### CODE REVIEW

**Comment by NicholasFlamy** [comment]
Pretty sure the e2e is inherited from #8228 and I need help with the server test.

**Comment by NicholasFlamy** [comment]
Undid some changes because they made it worse.

**Review by mertalev — APPROVED** [review]

**Comment by NicholasFlamy** [comment]
Yay, all but the e2e (which is failing the same as with your PR) are successful!

File: open-api/typescript-sdk/src/fetch-client.ts [hunk 1]
```diff
@@ -2852,6 +2857,10 @@ export enum AssetOrder {
     Asc = "asc",
     Desc = "desc"
 }
+export enum Role {
```
Comment (jrasm91) [inline open-api/typescript-sdk/src/fetch-client.ts:2860]: Why is this here?

Comment (NicholasFlamy) [reply open-api/typescript-sdk/src/fetch-client.ts:2860]: It was added by running `make open-api`.
